---
title: "Panicum hallii microbiome"
output: html_notebook
---

```{r}
library(tidyverse)
library(tidyMB)
library(vegan)
```

```{r}
ct <- readRDS("~/SMB/Albina/2017 Spring Microbiome seq/PH_Analysis/count_table.rds") %>% 
  t() %>% as.data.frame() %>% rownames_to_column("sequence") %>% as_tibble()
taxa <- readRDS("~/SMB/Albina/2017 Spring Microbiome seq/PH_Analysis/taxa.rds") %>% 
  as.data.frame() %>% rownames_to_column("sequence") %>% as_tibble()
map <- read.table("~/SMB/Albina/2017 Spring Microbiome seq/PH_Analysis/ph.map", header = T, sep = "\t")
info <- read.table("~/SMB/Albina/2017 Spring Microbiome seq/PH_Analysis/map.txt", header = T)
map2 <- map %>% inner_join(info, by = "SampleID") %>% 
  mutate(SampleID = gsub("\\.fastq\\.gz", "", RunID))
```

```{r}
ph_data <- ct %>% 
  gather(SampleID, value, -sequence) %>% 
  inner_join(taxa, by = "sequence") %>% 
  inner_join(map2, by = "SampleID") %>% 
  filter(Family != "Mitochondria" & Order != "Chloroplast" & Kingdom != "Eukaryota") %>% 
  group_by(SampleID) %>% 
  mutate(depth = sum(value), RA = (value / depth) * 1000, logRA = log2(RA + 1)) %>% 
  mutate(Phylum2 = NA) %>% 
  mutate(Phylum2 = ifelse(Phylum == "Proteobacteria", as.character(Class), as.character(Phylum))) %>% 
  group_by(sequence) %>% 
  nest() %>% 
  mutate(otu = paste("OTU", 1:nrow(.), sep = "")) %>% 
  unnest() %>% 
  select(-sequence) %>% 
  rename(Resource = Resourse)

write_rds(ph_data, path = "~/SMB/Albina/2017 Spring Microbiome seq/PH_Analysis/ph_data.rds")

ph_data <- read_rds("~/SMB/Albina/2017 Spring Microbiome seq/PH_Analysis/ph_data.rds")
```


```{r}
ph_data %>% 
  filter(Resourse == "nature") %>% 
  group_by(SampleID, Phylum2) %>% 
  summarise(phy_total = sum(RA)) %>% 
  group_by(Phylum2) %>% 
  nest() %>% 
  mutate(total = map_dbl(data, ~sum(.x$phy_total, na.rm = T))) %>% 
  top_n(10, total) %>% 
  unnest(data) %>% 
  ggplot(aes(SampleID, phy_total, fill = Phylum2)) +
  geom_bar(stat = "identity") +
  scale_fill_brewer(palette = "Spectral")
```

Let's try to get a feeling for what patterns are falling out of the data. We can do a PCoA to figure out if there are clustering patterns among variables in the data. 

```{r}
pPC <- tidy_pcoa(ph_data %>% select(otu, Compartment, Genotype, Species, SampleID, Trt, logRA, Resource, Intensity), otus = "otu", keep_loadings = T, value = "logRA")
```
From what I remember, soil was collected from 2 sites. Hallii plants found growing there at the sites had their microbiomes collected and sequenced. Soil from the site was brought back to the greenhouse and used to inoculate plants. Let's first see if can see any patterns differentiating nature from the greenhouse.

```{r}
pPC$axes %>% 
  ggplot(aes(MDS1, MDS2, color = Resource)) +
  geom_hline(yintercept = 0, linetype = "dotted") +
  geom_vline(xintercept = 0, linetype = "dotted") +
  geom_point() +
  theme_minimal()

```

It looks like there are 3 groups: 1 big group on the left of the x-axis and there are 2 groups on the right of the x-axis, but there seems to be no correlation with where these samples came from. We would expect to see a cluster for the hal site and a cluster for the fil site and potentially one or two more clusters for the greenhouse experiment. 

Is it possible that somehow the tubes were mislabeled when they went to Esther?